### PR TITLE
 Add new starting scenario The Last Delivery 

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
 ```SUMMARY: Content " Add new starting scenario The Last Delivery "```

#### Purpose of change
Add a new starting scenario
Add the ability to specify vehicles in scenarios
Expand Foodplace's lore a tiny bit

#### Describe the solution
Add starting_vehicle entry to scenarios
Add UI support to display starting vehicles
Add new vehicle, new quest and new food item to make the scenario

#### Testing
Chose `The Last Delivery`
Spawn properly
Get the vehicle
Thing Spawn properly in it

#### Additional context
Sometimes the food truck will spawn embedded into other vehicles, I don't know how to fix that.

### CODE REVIEW

File: data/json/npcs/missiondef.json [hunk 1]
```diff
@@ -577,6 +577,17 @@
     "start": "find_safety",
     "origins": [ "ORIGIN_NULL" ]
   },
+  {
+    "id": "MISSION_LAST_DELIVERY",
+    "type": "mission_definition",
+    "name": { "str": "Reach The Mansion And Finnish This Delivery" },
```
Comment (LaVeyanFiend) [inline data/json/npcs/missiondef.json:583]:
```suggestion
    "name": { "str": "Reach The Mansion And Finish This Delivery" },
```
_Insert joke about Finland not existing here_

File: data/json/items/comestibles/other.json [hunk 2]
```diff
@@ -449,6 +449,26 @@
     "vitamins": [ [ "calcium", 4 ], [ "iron", 47 ], [ "vitC", 7 ] ],
     "fun": -10
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "foodplace_food",
+    "name": { "str": "Foodplace's delicious food™", "str_pl": "Foodplace's delicious food™" },
```
Comment (LaVeyanFiend) [inline data/json/items/comestibles/other.json:455]: Shouldn't all these words be capitalized, since it's a branded product?

**Comment by Night-Pryanik** [comment]
No platinum chip references?

File: data/json/items/comestibles/other.json (hunk 2 above)
Comment (Fris0uman) [reply data/json/items/comestibles/other.json:455]: I think I'll keep it like that.

**Comment by Fris0uman** [comment]
Nah, we have enough fallout references already.

**Comment by Amoebka** [comment]
Isn't this duplicating the functionality of https://github.com/CleverRaven/Cataclysm-DDA/pull/37314 ?

**Comment by Fris0uman** [comment]
Nop it adds to it, that PR allowed *professions* to to specify vehicles this allows *scenarios* to do the same.

**Comment by Night-Pryanik** [comment]
> Nah, we have enough fallout references already.

And what are they? Besides power armor, which, honestly, isn't specifically Fallout-y.

**Comment by LaVeyanFiend** [comment]
> And what are they?
1) Razorclaws(Mirelurks)
2) Mininukes(and formerly the Fat Man before that got obsoleted)

**Comment by Night-Pryanik** [comment]
1. Razorclaws are non-bipedal, unlike Mirelurks. Also there are lot of other sources of mutated crustaceans besides Fallout.
2. Mininuke is (was) a RL thing. Also IIRC in Cataclysm it was never referenced as a "Fat Man".

**Comment by Fris0uman** [comment]
The eyebot is also a Fallout reference (even though we changed the description to be a quadcopter drone) and there might be a few others. In any case I'm not really interested in putting a fallout reference in that scenario.

File: data/json/npcs/missiondef.json [hunk 3]
```diff
@@ -577,6 +577,17 @@
     "start": "find_safety",
     "origins": [ "ORIGIN_NULL" ]
   },
+  {
+    "id": "MISSION_LAST_DELIVERY",
+    "type": "mission_definition",
+    "name": { "str": "Reach The Mansion And Finish This Delivery" },
+    "goal": "MGOAL_GO_TO_TYPE",
+    "destination": "mansion_c3_north",
```
Comment (ZhilkinSerg) [inline data/json/npcs/missiondef.json:585]: Should be `mansion_c3` to work on terrains with all rotations (see #40185).

**Pull request merged**